src/test/java/TicTacToe/TicTacToeTest.java

```diff
@@ -0,0 +1,109 @@
+package TicTacToe;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import static org.mockito.Mockito.*;
+
+import org.mockito.MockitoAnnotations;
+
+public class TicTacToeTest {
+
+    @Mock
+    private WinLogic winLogic;
+
+    @Mock
+    private Board board;
+
+    @Mock
+    private UserInterface userInterface;
+
+    @Mock
+    private MovementLogic movementLogic;
+
+    @Mock
+    private Position position;
+
+    private TicTacToe ticTacToe;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        ticTacToe = new TicTacToe(board, winLogic, movementLogic, userInterface);
+    }
+
+    @Test
+    public void testGameLoopContinuesWhileGameNotWon() {
+        when(winLogic.isGameContinues()).thenReturn(true, true, false);
+        when(movementLogic.doAMove(anyChar())).thenReturn(position);
+        when(position.getRowNumber()).thenReturn(0);
+        when(position.getColumnNumber()).thenReturn(0);
+
+        ticTacToe.loop();
+
+        verify(userInterface, times(1)).startGame();
```

Comment:
Rozumiem czmeu dajesz `veirfy`, ale osobiście nei ejstem fanem. Jakakolwiek drobna zmaina w kodzie bedzie wymagała tu poprawek. Skoncentrowałbym się na tym, żeby movementLogic wykonał 2 ruchy i to byłby moim zdnaiem wystarczający test.